File: moduletemplate/src/main/java/com/munch/module/template/mvp/IView.java
package com.munch.module.template.mvp;

import android.content.Context;
import androidx.annotation.Nullable;
import com.munch.module.template.rxjava.IDisposableManager;

/**
 * Created by Munch on 2019/7/29 17:13
 */
public interface IView<P extends IPresenter> {

    P bindPresenter(P p);

    @Nullable
    P getPresenter();

    Context getContext();

    IDisposableManager getManager();
}

File: common_base/src/main/java/com/munch/common/base/mvp/IView.kt
package com.munch.common.base.mvp

import android.content.Context

/**
 * Created by Munch on 2018/12/8.
 */
interface IView {

    /**
     * 如果非Context的子类须重写此方法
     */
    fun getContext(): Context {
        return this as Context
    }
}